Makefile

```diff
@@ -131,11 +131,12 @@ endif
 # Figure out optimize level.
 ifneq ($(DEBUG_LEVEL), 2)
```

Comment:
Are you allowing/expecting OPTIMIZE_LEVEL to be set when DEBUG_LEVEL=2?  This current code allows it.  

This might read cleaner as:
ifneq($(DEBUG_LEVEL,2)
ifneq ($OPTIMIZE_LEVEL,)
OPT += $(OPTIMIZE_LEVEL)
elif ($LITE)
else

Reply:
 I think it's conceptually simpler to allow `OPTIMIZE_LEVEL` (a more specific option) to override the level implied by `DEBUG_LEVEL` (a more general option). That doesn't mean all combinations will work, but if somebody specifies something like `OPTIMIZE_LEVEL=-O2`, I'd rather let it fail than sanitize it to something else.

Reply:
I am fine with falling through, but was wondering if that should be the intent or if -O0 should be set for debug (or if no optimization level should be set).  Please just add a comment as to the intent so it becomes clear.

Also, do you expect that the DEBUG_LEVEL=1 and 0 should use the same optimization level or not?

Reply:
> I am fine with falling through, but was wondering if that should be the intent or if -O0 should be set for debug (or if no optimization level should be set).

I am ok with leaving it blank and allowing the compiler default (`-O0` for gcc and clang) to be used. 

> Also, do you expect that the DEBUG_LEVEL=1 and 0 should use the same optimization level or not?

It seems ok. I usually think of `DEBUG_LEVEL=1` as enabling assertions (i.e., not specifying `-DNDEBUG`), while allowing optimizations.